diff --git a/docroot/sites/all/modules/nopremium/nopremium.module b/docroot/sites/all/modules/nopremium/nopremium.module
index 89da9bc..5af6bf2 100644
--- a/docroot/sites/all/modules/nopremium/nopremium.module
+++ b/docroot/sites/all/modules/nopremium/nopremium.module
@@ -227,6 +227,15 @@ function nopremium_access_full_content($node, $account = NULL) {
   if (empty($account)) {
     $account = $GLOBALS['user'];
   }
+  
+  // Allow other modules to provide access as well.
+  $results = module_invoke_all('nopremium_access', $node, $account);
+  foreach($results as $result) {
+    if ($result) {
+      return TRUE;
+    }
+  }
+
 
   return user_access('administer nodes', $account)
     || user_access('view full '. $node->type .' premium content', $account)
